src/python/zquantum/qaoa/parameter_initialization.py

```diff
@@ -0,0 +1,161 @@
+from typing import List, Tuple
+
+import numpy as np
+from zquantum.core.interfaces.cost_function import ParameterPreprocessor
+
+
+def _get_param(params_vector: List[float], layer_number):
+    """0 <= layer_number <= p + 1"""
+    if layer_number == 0 or layer_number == len(params_vector) + 1:
+        return 0
+    else:
+        return params_vector[layer_number - 1]
+
+
+def _perform_1_interpolation(
+    old_gammas: List[float], old_betas: List[float], p: int
+) -> Tuple[List[float], List[float]]:
+    """Following eq (B1) in the original paper, p is the length of old params."""
+    assert len(old_gammas) == len(old_betas) == p
+    new_gammas = []
+    new_betas = []
+
+    for i in range(1, p + 2):
+        # See eq (B1) in the original paper.
+        bef = i - 1
```

Comment:
`bef` is just the qubit before the current qubit `i` and it wasn't in the paper. I think I put it there because there was a lot of `i - 1`s. Would it be clearer just to get rid of the variable and just use `i - 1` where it shows up?

Reply:
Yes, `i-1` will be much easier to read.

Reply:
Done!